vehicle/src/Vehicle/Compile/Type/System.hs

```diff
@@ -39,6 +39,13 @@ class (Eq builtin, Hashable builtin, NormalisableBuiltin builtin, TypableBuiltin
     Type builtin ->
     m (Type builtin)
 
+  restrictRecordType ::
```

Comment:
Technically we're restricting the fields not the type. Maybe `restrictRecordAnnotatedAsTensor`?